tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,104 @@
+from unittest.mock import call
+import pytest
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+from ansible_collections.arista.cvp.plugins.module_utils.resources.modules.fields import ModuleOptionValues
+from tests.data.device_tools_unit import (validate_router_bgp,
+    validate_intf, validate_true, device_data, EXP_WARN_WARN, EXP_ERROR, EXP_VALID)
+
+@pytest.fixture
+def setup(apply_mock, mock_cvprac):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_configlet_info = apply_mock(TestValidateConfig.MOCK_LIST)
+    dummy_cvprac, mock_cvpClient = mock_cvprac
+    mock_cvpClient.api.validate_config_for_device.side_effect = dummy_cvprac.validate_config_for_device
+    user_topology = DeviceInventory(data=device_data)
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module)
+    return mock_ansible_module, mock__get_configlet_info, user_topology, cv_tools
+
+@pytest.mark.generic
+class TestValidateConfig():
+    """
+    Contains unit tests for validate_config()
+    """
+
+    # list of paths to patch
+    MOCK_LIST = [
+        'ansible_collections.arista.cvp.plugins.module_utils.device_tools.AnsibleModule',
+        'ansible_collections.arista.cvp.plugins.module_utils.device_tools.CvDeviceTools._CvDeviceTools__get_configlet_info']
+
+    @pytest.mark.parametrize("validate_mode, expected_calls",
+        [pytest.param(
+            ModuleOptionValues.VALIDATE_MODE_STOP_ON_WARNING, [call.fail_json(msg=EXP_WARN_WARN)], id="warning_stop_on_warning"
+        ),
+        pytest.param(
+            ModuleOptionValues.VALIDATE_MODE_STOP_ON_ERROR, [call.exit_json(msg=EXP_WARN_WARN)], id="warning_stop_on_error"
+        ),
+        pytest.param(
+            ModuleOptionValues.VALIDATE_MODE_IGNORE, [call.exit_json(msg=EXP_WARN_WARN)], id="warning_skip"
+        ),
+        ])
+    def test_validate_config_warning(self, setup, validate_mode, expected_calls):
+        """
+        warning case with
+        mode=[stop_on_warning, stop_on_error, ignore]
+        expected to fail
+        """
+        # call setup() to apply mocks/patches
+        mock_ansible_module, mock__get_configlet_info, user_topology, cv_tools = setup
+        # mock the return value of CvDeviceTools.get_configlet_info()
+        mock__get_configlet_info.return_value=validate_intf
```

Comment:
Looking at the pattern of the tests here, it could be better to have a way to give to setup the return_value for mock__het_configlet_info so that it can be done in the setup and abstracted from the test even further, this would in particular allow to implify even further the tests.

Also I dont think you need to get the mock_ansible_module returned as you can get it from cv_tools using

```
cv_tools.__ansible
```

as per https://github.com/aristanetworks/ansible-cvp/blob/devel/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py#L354

Reply:
You surely can, but I think for readability sake, I wanted to keep it separate.
On mock_ansible_module: AFAIK, you have to patch where the object is looked up and not where it's defined. 
https://docs.python.org/3/library/unittest.mock.html#where-to-patch
Here, I want to patch `AnsibleModule` that is imported in `device_tools`